include/picongpu/plugins/radiation/Radiation.kernel

```diff
@@ -347,85 +335,67 @@ namespace picongpu
-
-                                /* Particle loop: thread runs through loaded particle data
-                                 *
-                                 * Summation of Jackson radiation formula integrand
-                                 * over all electrons for fixed, thread-specific
-                                 * frequency
-                                 *
-                                 * This is the hot loop of this kernel, taking ~95% of its time.
-                                 * So performance-wise the rest of this kernel does not significantly contribute.
-                                 * On the contrary, few operations done inside the loop matter a lot.
-                                 */
-                                for(int j = 0; j < counter_s; ++j)
-                                {
-                                    // check Nyquist-limit for each particle "j" and each frequency "omega"
-                                    if(lowpass_s[j].check(omega))
+                                auto forEachOmega = lockstep::makeForEach<radiation_frequencies::N_omega>(worker);
+                                forEachOmega(
+                                    [&](uint32_t const o)
                                     {
-                                        // Here happens the true physical calculation
-                                        float_X const formFactorValue = myRadFormFactor(radWeighting_s[j]);
-                                        float_X const phase = t_ret_s[j] * omega;
-                                        amplitude.addContribution(real_amplitude_s[j], phase, formFactorValue);
-                                    }
-                                }
-
-                                /* the radiation contribution of the following is added to global memory:
-                                 *     - valid particles of last super cell
-                                 *     - from this (one) time step
-                                 *     - omega_id = theta_idx * radiation_frequencies::N_omega + o
-                                 */
-                                radiation(DataSpace<2>(theta_idx * radiation_frequencies::N_omega + o, jobIdx))
-                                    += amplitude;
-
-                            } // end frequency loop
-
-
-                            // wait till all radiation contributions for this super cell are done
-                            worker.sync();
-
-                            /* First threads starts loading next frame of the super-cell:
-                             *
-                             * Info:
-                             *   The calculation starts with the last SuperCell (must not be full filled)
-                             *   all previous SuperCells are full with particles
-                             */
-                            particlesInFrame = frameSize;
```

Comment:
Is this now part of the `forEachParticleInFrame` loop?

Reply:
Yes this is all hidden in the for each.